v0/destinations/am/transform.js

```diff
@@ -327,6 +324,37 @@ function responseBuilderSimple(
     default:
       if (message.channel === "mobile") {
         set(payload, "device_brand", message.context.device.manufacturer);
+
```

Comment:
You can check the value `context.device.adTrackingEnabled` initially and decide whether you need to set the `idfa`/`idvf`/`adid` in the final payload. 

Also, for getting the values use the `get` method from the `get-value` library. It handles the `null/undefined` well. Otherwise, statements like `message.context.device.type` might cause potential throw and discard the entire event.

Reply:
Mentioned changes are added